daemon/cluster/provider/cluster.go

```diff
@@ -0,0 +1,36 @@
+package provider
```

Comment:
Do you have any suggestion on how to call this interface and where to put it?
Libnetwork is only a consumer of it, and the cluster in Moby is the one that implements it.